gazelle_haskell_modules/lang.go

```diff
@@ -213,26 +213,37 @@ func (*gazelleHaskellModulesLang) Fix(c *config.Config, f *rule.File) {
 				r.Delete()
 			}
 		}
-		fixModuleLists(r, ruleNameSet)
+		fixModulesList(r, ruleNameSet)
 	}
 
 	f.Sync()
 }
 
-// Leaves only those modules from r that are in ruleNameSet.
+func fixModulesList(r *rule.Rule, ruleNameSet map[string]bool) {
+	// TODO: use labels instead of manually stripping away the ':' ?
+	stripColon := func(module string) string { return module[1:] }
+	fixModulesLists(r, ruleNameSet, "modules", stripColon)
+}
+
+// Leaves only those modules from r that are in ruleNameSet according to toRuleName.
 // If a module is not in ruleNameSet, that means that there was no info associated with it,
 // hence it should be deleted.
-func fixModuleLists(r *rule.Rule, ruleNameSet map[string]bool) {
-	modules := r.AttrStrings("modules")
+//
+// modulesFieldName is expected to be "modules" or "hidden_modules".
```

Comment:
Upon trying to applies this, I feel like it makes the code less clear. It is true that we're doing string manipulation here, but we still need to keep in mind these are specific kinds of strings. I also agree that calling the passed thing "modules" is a bit misleading, since they need some string manipulation to become the names of module rules.

Is there some middle ground between these two?

Reply:
The rationale for my suggestion is that the code doesn't care what the string is, as long as the output of toRuleName is a rule name that might appear in the set. But I don't feel strong about it. We can keep the names as they are.

Reply:
I added this bit as part of e18e2de
```
// The field is removed if all the strings are dropped from the list.
```